pipeline/frontend/yaml/compiler/option.go

```diff
@@ -143,6 +143,14 @@ func WithProxy() Option {
 	)
 }
 
+// WithNetworks configures the compiler with additionnal networks
+// to be connected to build containers
+func WithNetworks() Option {
```

Comment:
I recommend `func WithNetworks(networks []string) Option` so that the application importing this library can choose the name of the environment variables and pass the information accordingly.